### PR TITLE
[xbmc] Allow enter an user domain in GUI  for SMB protocol.

### PR DESCRIPTION
**Pull request by afedchin**
see title. 

currently it's not possible in GUI somehow define domain for an user. this introduces an ability to enter user name in format 'DOMAIN\user' or `DOMAIN/user` and this will be properly handled.

### CODE REVIEW

File: xbmc/filesystem/Directory.cpp [hunk 1]
```diff
@@ -214,31 +215,37 @@ bool CDirectory::GetDirectory(const CURL& url, CFileItemList &items, const CHint
       // hide credentials if necessary
       if (CPasswordManager::GetInstance().IsURLSupported(realURL))
       {
-        for (int i = 0; i < items.Size(); ++i)
+        bool hide = false;
+        // for explicitly credetials
```
Comment (notspiff) [inline xbmc/filesystem/Directory.cpp]: credentials

File: xbmc/filesystem/Directory.cpp [hunk 2]
```diff
@@ -214,31 +215,37 @@ bool CDirectory::GetDirectory(const CURL& url, CFileItemList &items, const CHint
...
+        // for explicitly credetials 
+        if (!realURL.GetUserName().empty())
         {
-          CFileItemPtr item = items[i];
-          CURL itemUrl = item->GetURL();
-          // for explicitly credetials 
-          if (!realURL.GetUserName().empty())
+          // credentials was changed i.e. were stored in the password
+          // manager, in this case we can hide them from an item URL,
+          // otherwise we have to keep cretendials in an item URL
```
Comment (notspiff) [inline xbmc/filesystem/Directory.cpp]: credentials

File: xbmc/filesystem/SMBFile.cpp [hunk 3]
```diff
@@ -239,16 +239,15 @@ std::string CSMB::URLEncode(const CURL &url)
...
-  {
-    flat += URLEncode(url.GetDomain());
-    flat += ";";
-  }
-
   /* samba messes up of password is set but no username is set. don't know why yet */
   /* probably the url parser that goes crazy */
   if(url.GetUserName().length() > 0 /* || url.GetPassWord().length() > 0 */)
   {
+    if(url.GetDomain().length() > 0)
```
Comment (notspiff) [inline xbmc/filesystem/SMBFile.cpp]: ! .empty()

**Review by notspiff — APPROVED** [review]
just some (inherited) cosmetics, looks good.

**Comment by afedchin** [comment]
oh, sorry, didn't see comments. will fix them in next PR.

**Pull request merged**